File: src/Test/Yield.js
class TYield {
  *Generator1() {
    for (let i = 0; i < 10; i++) {
      yield i;
    }
  }

  *Generator2() {
    const values = ['one', 'two', 'three'];
    for (let i = 0; i < values.length; i++) {
      yield values[i];
    }
  }
}

const Yield = new TYield();

for (const x of Yield.Generator1()) {
  console.log(x);
}

for (const x of Yield.Generator2()) {
  console.log(x);
}

File: src/Test/Yield.py
class TYield():
    def Generator1(self):
        for i in range(10):
            yield i

    def Generator2(self):
        for i in ['one', 'two', 'three']:
            yield i



Yield = TYield()
for x in Yield.Generator1():
    print(x)

for x in Yield.Generator2():
    print(x)
